tests/Fixtures/Integration/misc/PHP8_2.test

```diff
@@ -101,6 +101,12 @@ trait WithConstants
     private const THREE = 'three';
 }
 
+// https://wiki.php.net/rfc/dnf_types
+function generateSlug((HasTitle&HasId)|null $post)
```

Comment:
I believe this integration test is too optimistic. I've tested my own data set from #6608 ([comment](https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/issues/6608#issuecomment-1241958302)) against your branch and test fails:

<img width="1728" alt="image" src="https://user-images.githubusercontent.com/600668/223250426-187e7fea-524a-4704-b5b8-7db67d54d0c7.png">

Input code is:

```php
<?php

interface DnfA
{
    public function foo(): void;

    public function bar(): void;

    public function baz(): void;
}
interface DnfB
{
    public function foo(): void;
}
interface DnfC
{
    public function bar(): void;
}
interface DnfD
{
    public function baz(): void;
}
interface DnfE extends DnfB
{
}
interface DnfF extends DnfC
{
}
interface DnfG extends DnfD
{
}

class DnfTypes {
    private DnfA | ( DnfB & DnfC & DnfD) | ( DnfE & DnfF & DnfG ) $whatever;

    public function __construct(
        private DnfA | ( DnfB & DnfC & DnfD) | ( DnfE & DnfF & DnfG ) $notSure
    ) {
    }

    public function notSure(): DnfA | ( DnfB & DnfC & DnfD) | ( DnfE & DnfF & DnfG )
    {
        return $this->notSure;
    }

    public function setWhatever(DnfA | ( DnfB & DnfC & DnfD) | ( DnfE & DnfF & DnfG ) $whatever): void
    {
        $this->whatever = $whatever;
    }
}
```

and it does not contain syntax errors:

```shell
docker run -it --rm -v $PWD/dnf.php:/dnf.php php:8-cli-alpine sh -c "php --version && php -l /dnf.php"
PHP 8.2.3 (cli) (built: Feb 14 2023 21:12:31) (NTS)
Copyright (c) The PHP Group
Zend Engine v4.2.3, Copyright (c) Zend Technologies
No syntax errors detected in /dnf.php
```

Linting exception is thrown _after_ applying the fix, in `PhpCsFixer\Runner\Runner#216`, because the fixed `DnfTypes` class looks like:

```php
class DnfTypes
{
    private DnfA|(DnfB&DnfC&DnfD)|(DnfE&DnfF&DnfG) public $whatever;

    public function __construct(
        private DnfA|(DnfB&DnfC&DnfD)|(DnfE&DnfF&DnfG) $notSure
    ) {
    }

    public function notSure(): DnfA|(DnfB&DnfC&DnfD)|(DnfE&DnfF&DnfG)
    {
        return $this->notSure;
    }

    public function setWhatever(DnfA|(DnfB&DnfC&DnfD)|(DnfE&DnfF&DnfG) $whatever): void
    {
        $this->whatever = $whatever;
    }
}
```

Took me a while to realise the problem, but look at `private DnfA|(DnfB&DnfC&DnfD)|(DnfE&DnfF&DnfG) public $whatever;` → there's `private [...] public $whatever`, visibility modifier is provided twice.

I believe integration test `PHP8_2.test` needs improvements in terms of DNF with more complex usages and problem mentioned above must be fixed 🙂

Reply:
This seems to be a bug in VisibilityRequiredFixer, not related to changes from this PR (present at master branch too).